src/Avalonia.Controls/DateTimePickers/TimePicker.cs

```diff
@@ -63,63 +59,73 @@ public class TimePicker : TemplatedControl
         private Grid? _contentGrid;
         private Popup? _popup;
 
-        private TimeSpan? _selectedTime;
-        private int _minuteIncrement = 1;
-        private string _clockIdentifier = "12HourClock";
-
         public TimePicker()
         {
             PseudoClasses.Set(":hasnotime", true);
 
             var timePattern = CultureInfo.CurrentCulture.DateTimeFormat.ShortTimePattern;
             if (timePattern.IndexOf("H") != -1)
-                _clockIdentifier = "24HourClock";
+                SetCurrentValue(ClockIdentifierProperty, "24HourClock");
+        }
+
+        static TimePicker()
+        {
+            MinuteIncrementProperty.Changed.AddClassHandler<TimePicker>((s, e) => s.SetSelectedTimeText());
+
+            ClockIdentifierProperty.Changed.AddClassHandler<TimePicker>((s, e) =>
+            {
+                s.SetGrid();
+                s.SetSelectedTimeText();
+            });
+
+            SelectedTimeProperty.Changed.AddClassHandler<TimePicker, TimeSpan?>((s, e) => {
+                s.OnSelectedTimeChanged(e.OldValue.Value, e.NewValue.Value);
+                s.SetSelectedTimeText();
+            });
```

Comment:
Virtual OnPropertyChanged will be nicer here, as AddClassHandler inernally will create an observer for the observable, which internally will listen for the same PropertyChanged event, just with observable layer in between.

Reply:
I recall you making this point in another PR recently, but it's not reflected here because I wrote most of this code over a month ago.

It sounds a lot like `AddClassHandler` could use a round of optimisation, as it's quite counter-intuitive that globbing everything together in a single virtual override for all property changes is more efficient than registering small, specialised handlers.

There are also two benefits to the subscriber approach: 
* A more derived type can't block the handler from running (accidentally or otherwise).
* It's easier to discover the handler if it's defined next to the property, instead of in the middle of one potentially massive method somewhere else. Ideally it would be provided as an argument to the `Register` method itself.

Okay, so in _this_ case they are in the static constructor...but mostly I kept the lines of code in the same place and just wrapped it in a method body. This avoids the lines moving in the file and losing their source control history.

Anyway, despite the above I am willing to make the change if you feel strongly about this.

Reply:
In the past we have slowly moving to OnPropertyChange, so yes, it would make sense.

Reply:
I've looked at the code behind `AddClassHandler`, and it's a lost cause. I'm now fully on board with this change! 👍

Reply:
Yea, I switch to OnPropertyChanged with all controls I refactor too. I would argue it improves readability in most cases too because the change handling methods are small anyway.